drv/fpga-server/Cargo.toml

```diff
@@ -7,19 +7,26 @@ edition = "2021"
 userlib = {path = "../../sys/userlib", features = ["panic-messages"]}
 ringbuf = {path = "../../lib/ringbuf" }
 gnarle = {path = "../../lib/gnarle"}
+drv-fpga-api = {path = "../../drv/fpga-api"}
+drv-fpga-devices = {path = "../../drv/fpga-devices", features = ["front_io"]}
+drv-i2c-api = {path = "../../drv/i2c-api", optional = true}
+drv-i2c-devices = {path = "../../drv/i2c-devices", optional = true}
 drv-spi-api = {path = "../../drv/spi-api"}
 drv-stm32xx-sys-api = {path = "../../drv/stm32xx-sys-api", features = ["family-stm32h7"]}
-drv-fpga-api = {path = "../../drv/fpga-api"}
-drv-fpga-devices = {path = "../../drv/fpga-devices"}
 idol-runtime = {git = "https://github.com/oxidecomputer/idolatry.git"}
 cfg-if = "1"
 num-traits = { version = "0.2.12", default-features = false }
 zerocopy = "0.6.1"
 
 [build-dependencies]
+build-i2c = {path = "../../build/i2c"}
 build-util = {path = "../../build/util"}
 idol = {git = "https://github.com/oxidecomputer/idolatry.git"}
 
+[features]
+mainboard = []
```

Comment:
Since these are mutually exclusive, there should be a guard in `drv/fpga-server/src/main.rs`, e.g.
```rust
#[cfg(all(target_board = "sidecar-1", feature = "mainboard", feature = "front_io"))]
compile_error!("Cannot enable both mainboard and front_io simultaneously");
```